File: agent-plugin/plugin/src/main/java/org/matilda/java/services/reflection/ProxyHandlerService.java
package org.matilda.java.services.reflection;

import org.matilda.commands.MatildaCommand;
import org.matilda.commands.MatildaDynamicService;
import org.matilda.java.services.reflection.protobuf.JavaValue;

import java.util.List;

@MatildaDynamicService
public interface ProxyHandlerService {
    @MatildaCommand
    JavaValue invoke(long methodId, List<JavaValue> argumentIds);
}

File: agent-plugin/plugin/src/main/java/org/matilda/java/services/reflection/ProxyHandlerService.kt
package org.matilda.java.services.reflection

import org.matilda.commands.MatildaCommand
import org.matilda.commands.MatildaDynamicService
import org.matilda.java.services.reflection.protobuf.JavaValue

@MatildaDynamicService
fun interface ProxyHandlerService {
    @MatildaCommand
    operator fun invoke(methodId: Long, argumentIds: List<JavaValue>): JavaValue
}
